instagram/src/Components/Authentication/Authenticate.1.js

```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import Login from "../Login/Login";
+
+export default PassedComponent => {
+  return class AuthenticationClass extends React.Component {
+    componentDidUpdate() {
+      if (!this.props.credentials && localStorage.getItem("credentials")) {
```

Comment:
isn't this logic redundant?  Because this.props.credentials was set to localStorage.getItem('credentials') in the parent component so the left and right of the `&&` are always the same value. Not sure about this.